BaseTools/Plugin/OverrideValidation/OverrideValidation.py

```diff
@@ -613,6 +616,67 @@ def ModuleHashCal(path):
     result = hash_obj.hexdigest()
     return result
 
+
+# This calculates the md5 for the specified path.
+# If the path is an INF, it hashes the sources section
+# If its a directory, all the files in the directory and
+# sub directories will be hashed.
+# this version will sort the files prior to generating the hashes
+# due to some errors encountered under linux where the file order
+# returned from os.walk were different, causing a hash mismatch
+def ModuleHash2Cal(path):
+
+    sourcefileList = []
+    binaryfileList = []
+    hash_obj = hashlib.md5()
+
+    # Find the specific line of Sources section
+    folderpath = os.path.dirname(path)
+
+    if os.path.isdir(path):
+        # Collect all files in this folder to the list
+        for subdir, _, files in os.walk(path):
+            for file in files:
+                sourcefileList.append(os.path.join(subdir, file))
+    else:
+        sourcefileList.append(path)
+
+    if path.lower().endswith(".inf") and os.path.isfile(path):
+
+        # Use InfParser to parse sources section
+        ip = InfParser()
+        ip.ParseFile(path)
+
+        # Add all referenced source files in addition to our inf file list
+        for source in ip.Sources:
+            sourcefileList.append(os.path.normpath(os.path.join(folderpath, source)))
+
+        # Add all referenced binary files to our binary file list
+        for binary in ip.Binaries:
+            binaryfileList.append(os.path.normpath(os.path.join(folderpath, binary)))
+
+    # Convert to '/' path separator prior to sorting to avoid ordering mismatches
+    sourcefileList = [sfile.replace("\\", "/") for sfile in sourcefileList]

```

Comment:
did something like

`sourcefileList = [pathlib.Path(sfile).as_posix() for sfile in sourcefileList]`

now work well enough?

Reply:
that worked, I updated to include that. 

Now on to further testing.